src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -341,6 +341,16 @@ Domain<3> BinaryCompactObject::create_domain() const {
       3, std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>;
 
   std::vector<BcMap> boundary_conditions_all_blocks{};
+  // Add an empty map to the boundary conditions for blocks that have no
+  // external boundaries
```

Comment:
Could you add a brief "why" like `. Factored to avoid code duplication.` or something? I was confused that you needed this function since I hadn't looked at the domain creators in a while